pkg/flare/archive_dca.go

```diff
@@ -176,3 +180,84 @@ func getDCAWorkloadList() ([]byte, error) {
+	}
+	agentDaemonset, err = GetDaemonset(cl, agentDaemonsetName, releaseNamespace)
+	if err != nil {
+		return log.Errorf("Error while collecting the Agent DaemonSet: %q", err)
+	}
+	return fb.AddFile("agent-daemonset.yaml", agentDaemonset)
+}
+
+// getClusterAgentDeployment retrieves the Deployment manifest of the Cluster Agent
+func getClusterAgentDeployment(fb flarehelpers.FlareBuilder) error {
+	// The Cluster Agent Deployment name is based on the Helm chart template and added to the Cluster Agent as an environment variable
+	var clusterAgentDeploymentName string
+	var releaseNamespace string
+	var clusterAgentDeployment []byte
+
+	cl, err := apiserver.GetAPIClient()
+	if err != nil {
+		return log.Errorf("Can't create client to query the API Server: %s", err)
+	}
+	clusterAgentDeploymentName = os.Getenv(HELM_CLUSTER_AGENT_DEPLOYMENT)
+	releaseNamespace = os.Getenv(HELM_CHART_RELEASE_NAMESPACE)
+	if clusterAgentDeploymentName == "" || releaseNamespace == "" {
+		return log.Errorf("Can't collect the Cluster Agent Deployment name and/or namespace from the environment variables %s and %v", HELM_CLUSTER_AGENT_DEPLOYMENT, HELM_CHART_RELEASE_NAMESPACE)
+	}
+	clusterAgentDeployment, err = GetDeployment(cl, clusterAgentDeploymentName, releaseNamespace)
+	if err != nil {
+		return log.Errorf("Error while collecting the Cluster Agent Deployment: %q", err)
+	}
+	return fb.AddFile("cluster-agent-deployment.yaml", clusterAgentDeployment)
+}
+
+// getHelmValues retrieves the user-defined values for the Datadog Helm chart
+func getHelmValues(fb flarehelpers.FlareBuilder) error {
+	var helmUserValues []byte
+	var releaseName string
+	var releaseNamespace string
+
+	cl, err := apiserver.GetAPIClient()
+	if err != nil {
+		return log.Errorf("Can't create client to query the API Server: %s", err)
+	}
+	releaseName = os.Getenv(HELM_CHART_RELEASE_NAME)
+	releaseNamespace = os.Getenv(HELM_CHART_RELEASE_NAMESPACE)
+	if releaseName == "" || releaseNamespace == "" {
+		return log.Errorf("Can't collect the Datadog Helm chart release name and/or namespace from the environment variables %s and %v", HELM_CHART_RELEASE_NAME, HELM_CHART_RELEASE_NAMESPACE)
+	}
+	helmUserValues, err = GetDatadogHelmUserValues(cl, releaseName, releaseNamespace, "secret")
+	if err != nil {
+		log.Warnf("Error while collecting the Helm chart values from secret: %v", err)
+	}
+	// The returned bytes array is not nil, we can exit
+	if len(helmUserValues) != 0 {
+		return fb.AddFile("helm-values.yaml", helmUserValues)
+	}
+	// The cluster Agent was unable to retrieve Helm chart data from secrets, attempting to retrieve them from Configmaps
+	helmUserValues, err = GetDatadogHelmUserValues(cl, releaseName, releaseNamespace, "configmap")
+	if err != nil {
+		log.Warnf("Error while collecting the Helm chart values from configmap: %v", err)
+	}
```

Comment:
thanks to all the refactoring maybe it is not needed anymore to have the `GetDatadogHelmUserValues()` function. IMO, the code of GetDatadogHelmUserValues() can be moved directly in `getHelmValues()`.